### PR TITLE
qemu.test: migration_multi_host_with_kdump

### PR DESCRIPTION
**Pull request by XueqiangWei**
1. add function "start_worker_guest_kdump"
2. add function "check_worker_kdump"

Signed-off-by: Xueqiang Wei xuwei@redhat.com

### CODE REVIEW

**Comment by XueqiangWei** [comment]
ID 1297298

**Comment by XueqiangWei** [comment]
09:55:04 INFO | 			GOOD	virt.case3.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.ping-pong-kdump.vhost_src_yes_dst_yes.mig_online.tcp	virt.case3.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.ping-pong-kdump.vhost_src_yes_dst_yes.mig_online.tcp	timestamp=1476438910	localtime=Oct 14 09:55:10	completed successfully
09:55:04 INFO | 		END GOOD	virt.case3.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.ping-pong-kdump.vhost_src_yes_dst_yes.mig_online.tcp	virt.case3.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.ping-pong-kdump.vhost_src_yes_dst_yes.mig_online.tcp	timestamp=1476438910	localtime=Oct 14 09:55:10	
09:55:04 INFO | 	END GOOD	----	----	timestamp=1476438910	localtime=Oct 14 09:55:10	
09:55:06 INFO | Client complete
09:55:15 INFO | 			GOOD	virt.case3.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.ping-pong-kdump.vhost_src_yes_dst_yes.mig_online.tcp	virt.case3.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.ping-pong-kdump.vhost_src_yes_dst_yes.mig_online.tcp	timestamp=1476438921	localtime=Oct 14 09:55:21	completed successfully
09:55:15 INFO | 		END GOOD	virt.case3.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.ping-pong-kdump.vhost_src_yes_dst_yes.mig_online.tcp	virt.case3.remote_nfs.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.multi_host.migrate_multi_host.ping-pong-kdump.vhost_src_yes_dst_yes.mig_online.tcp	timestamp=1476438921	localtime=Oct 14 09:55:21	
09:55:15 INFO | 	END GOOD	----	----	timestamp=1476438921	localtime=Oct 14 09:55:21

**Comment by suqinhuang** [comment]
your branch name is really long ...

**Comment by PyLearner** [comment]
lgtm
Acked-By : Meng Yang meyang@redhat.com

**Comment by vivianQizhu** [comment]
Acked-by: Qianqian Zhu qizhu@redhat.com

File: qemu/tests/migration_multi_host_with_kdump.py [hunk 1]
```diff
@@ -71,6 +72,70 @@ def __init__(self, test, params, env):
...
+                    if session is not None:
+                        logging.info("kdump completed, no need ping-pong"
+                                     " migration")
+                        self.stop_migrate = True
+                    output = session.cmd_output(vmcore_chk_cmd)
+                    kdump.postprocess_kdump(vm, self.login_timeout)
+                    if not output:
+                        raise error.TestFail("Could not found vmcore file")
+                    elif vmcore_incomplete in output.split("\n"):
+                        raise error.TestError("Kdump is failed")
```
Comment (CongLi) [inline qemu/tests/migration_multi_host_with_kdump.py]: TestError is for those env preparation failed scenarios.
So I think should use TestFail here.

File: qemu/tests/migration_multi_host_with_kdump.py [hunk 2]
```diff
@@ -71,6 +72,70 @@ def __init__(self, test, params, env):
...
+                        vm.resume()
+                    if not utils_test.qemu.guest_active(vm):
+                        raise error.TestFail("Guest not active "
+                                             "after migration")
+                    logging.info("Logging into migrated guest after "
+                                 "migration")
+                    session = vm.wait_for_login(timeout=self.login_timeout)
+                    error.context("Checking vmcore file in guest",
+                                  logging.info)
+                    if session is not None:
```
Comment (CongLi) [inline qemu/tests/migration_multi_host_with_kdump.py]: Where close this session?

File: qemu/tests/migration_multi_host_with_kdump.py (hunk 1 above)
Comment (XueqiangWei) [inline qemu/tests/migration_multi_host_with_kdump.py]: agree with you, will update

File: qemu/tests/migration_multi_host_with_kdump.py (hunk 2 above)
Comment (XueqiangWei) [inline qemu/tests/migration_multi_host_with_kdump.py]: thx, will add

**Comment by XueqiangWei** [comment]
@CongLi updated, please help review it, many thanks.

**Comment by CongLi** [comment]
LGTM,
acked-by: Cong Li <coli@redhat.com>

**Comment by CongLi** [comment]
Thanks @XueqiangWei @PyLearner @vivianQizhu .

**Pull request merged**